synthesis/codon/analyzer.go

```diff
@@ -0,0 +1,36 @@
+package codon
+
+import "github.com/TimothyStiles/poly/io/genbank"
+
+// Stats denotes a set of computed codon table statistics
+type Stats struct {
+	StartCodonCount map[string]int `json:"start_codons_counts"`
+}
+
+func NewStats() *Stats {
+	return &Stats{
+		StartCodonCount: map[string]int{},
+	}
+}
```

Comment:
Why define a new struct which has a single member and no methods?

This should either be a type alias (`type Stats map[string] int`) or not be a type at all. IMO the latter is preferable.

Reply:
I'd use type aliases more to enhance readabilty/make it harder to mess up (e.g `type DNA string`), this is meant as a container struct because we presumably will add more stats in the future. Rather than having a large list of `ints` and `maps` in a struct that has nothing to do with codon stats, it's nicer to have it in its own type.

Where would you put it if you'd like to keep it as an untyped map? `codonTable`?

Reply:
Unless there's a case where the stats will be used together often, I think they should be returned by different methods or functions.

IMO it should be something like the following would be best:

```
func ComputeStartCodonStatistics(codonTable Table) map[string]int {
  // code
  return statisticsMap
}

func ComputeAnotherStatistic(codonTable Table) SomeType {
  // code
  return instanceOfSomeType
}
```